proxy/coco/src/peer/run_state.rs

```diff
@@ -67,9 +78,26 @@ impl MaybeFrom<&Input> for Event {
             Input::Announce(input::Announce::Succeeded(updates)) => {
                 Some(Self::Announced(updates.clone()))
             },
-            Input::Peer(event) => Some(Self::Peer(event.clone())),
+            Input::Peer(event) => match event {
+                PeerEvent::GossipFetch(FetchInfo {
```

Comment:
Can we provide a `From` impl?

Reply:
From which type to which?

Reply:
`FetchInfo` to GossipFetched`, or even just re-use `FetchInfo` in the `Event`